lms/lmsweb/views.py

```diff
@@ -152,10 +152,12 @@ def exercises_page():
     fetch_archived = bool(request.args.get('archived'))
     exercises = Solution.of_user(current_user.id, fetch_archived)
     is_manager = current_user.role.is_manager
+    is_banned = current_user.role.is_banned
```

Comment:
This is coupled and creates code duplication. Please use Flask's `@webapp.before_request` instead.